### PR TITLE
Remove I2C and Update API and RDK in revolution pi module

### PR DESCRIPTION
**Pull request by randhid**
@thegreatco I may have fixed the build issues :)

### CODE REVIEW

File: revolution_pi/revolution_pi_nonlinux.go [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// This is a non-linux package implementation for the revolution pi.
+package revolution_pi
```
Comment (randhid) [inline revolution_pi/revolution_pi_nonlinux.go:2]: this file may not be necessary.

File: revolution_pi/revolution_pi.go [hunk 2]
```diff
@@ -12,22 +12,20 @@ import (
...
 	"go.viam.com/rdk/grpc"
+	"go.viam.com/rdk/logging"
 	"go.viam.com/rdk/resource"
 )
 
-var Model = resource.NewModel("viam-labs", "kunbus", "revolutionpi")
-
 type revolutionPiBoard struct {
 	resource.Named
+	resource.AlwaysRebuild
```
Comment (thegreatco) [inline revolution_pi/revolution_pi.go:25]: What exactly does this do?

File: revolution_pi/config.go [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
 
 import (
+	"go.viam.com/rdk/resource"
 	"go.viam.com/rdk/utils"
 )
 
+var Model = resource.NewModel("viam-labs", "kunbus", "revolutionpi")
+
 type Config struct {
+	resource.TriviallyValidateConfig
```
Comment (thegreatco) [inline revolution_pi/config.go:11]: To be honest, we may want to revisit this module's config in light of how I did the modbus module. In the modbus module I take a map of pin names to addresses. We might want to do that here too.

File: revolution_pi/revolution_pi.go (hunk 2 above)
Comment (randhid) [reply revolution_pi/revolution_pi.go:25]: It always triggers a reconstruction on a config change. In light of your next comment, I think this is the wrong call. What does your config look like? paste it here.

Comment (randhid) [reply revolution_pi/revolution_pi.go]: Changed it to `triviallyReconfigure`, which is what the old reconfigure function was doing.

File: revolution_pi/config.go (hunk 3 above)
Comment (randhid) [reply revolution_pi/config.go:11]: We can do that in a follow up - I might have eng time for you on my plan for this module next cycle.

**Comment by randhid** [comment]
Can I merge this in or no?

**Pull request merged**